tests/integration/startup.test.js

```diff
@@ -1,189 +1,152 @@
 const {exec, spawn} = require('child_process');
+const {readDaemonFile, writeDaemonFile, getDaemonOutputDir} = require('../../src/FileService');
+const {generateSwarm} = require('../../src/daemonManager');
 
-const {despawnSwarm, clearDaemonStateAndConfigs} = require('../../utils/daemon/setup');
-const {editFile, generateSwarmJsonsAndSetState} = require('../../utils/daemon/configs');
 
+describe('daemon startup', function () {
 
-describe('daemon startup', () => {
+    const DAEMON_OBJ = {
+        listener_port: 50000,
+        directory_name: 'daemon-50000',
+        config_name: 'bluzelle-50000.json'
+    };
 
-    beforeEach('generate configs and set harness state', async () =>
-        await generateSwarmJsonsAndSetState(1));
-
-    afterEach('remove configs and peerslist', () => {
-        clearDaemonStateAndConfigs();
+    beforeEach('generate configs and set harness state', function () {
+        this.swarm = generateSwarm({numberOfDaemons: 1});
     });
 
-    describe('cmd line', () => {
+    afterEach('remove configs and peerslist and clear harness state', async function () {
+        await this.swarm.stop();
+        this.swarm.removeSwarmState();
+    });
 
-        context('accepts flags', () => {
+    describe('cmd line', function () {
 
-            it('accepts -h', async () => new Promise((resolve) => {
+        context('accepts flags', function () {
 
-                exec('cd ./daemon-build/output/; ./swarm -h', (error, stdout, stderr) => {
+            it('accepts -h', function (done) {
+                exec(`cd ${getDaemonOutputDir(DAEMON_OBJ)}; ./swarm -h`, (error, stdout, stderr) => {
```

Comment:
While you're refactoring, you may want to prevent additional refactors in the future by abstracting this a bit more.

Previously:
```
exec('cd ./daemon-build/output/; ./swarm -h', (error, stdout, stderr) => {...}
```

Current:
```
exec(`cd ${getDaemonOutputDir(DAEMON_OBJ)}; ./swarm -h`, (error, stdout, stderr) => {...}
```

Proposed:
```
launchSwarm('-h', (err, stdout, stderr) => {...}
```

The benefit of the proposed approach is that it fully abstracts how to launch `swarm` except for passing arguments. That way, if the way to launch (minor example, abs path instead of cd then ./swarm) `swarm` changes in the future, it can be updated in one place instead of having to do a multi-file multi-line refactor again.

Additionally, you could do extra things in `launchSwarm` like add logging, or handle `error` directly (add information, raise informative exception/logging) and only pass along `stdout` and `stderr`. However, you should always have a way to get at `error` for negative testing (e.g. assert certain exit codes given certain invocations).

Thoughts?

Reply:
Good call, definitely should have done that in the first place. I've added this.